src/pkg/cli/client/byoc/do/byoc.go

```diff
@@ -644,13 +644,32 @@ func (b *ByocDo) update(service composeTypes.ServiceConfig) *defangv1.ServiceInf
 	return si
 }
 
-func (b *ByocDo) setUp(ctx context.Context) error {
+func (b *ByocDo) needsSetup(ctx context.Context) (bool, error) {
 	projectCdImageTag, err := b.getCdImageTag(ctx)
+
 	if err != nil {
-		return err
+		if strings.Contains(err.Error(), "no bucket found") {
+			return true, nil
+		} else {
+			return true, err
+		}
 	}
 
+	b.cdImageTag = projectCdImageTag
 	if b.SetupDone && b.cdImageTag == projectCdImageTag {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func (b *ByocDo) setUp(ctx context.Context) error {
+	needsSetup, err := b.needsSetup(ctx)
```

Comment:
Could you clarify what you mean by "this logic"? We're trying to load the cd image tag here, so I'm not sure how the task definition is related.

Reply:
The AWS code has a `setUp` function that creates the task definitions and has to be re-run if the tag changes, because that's part of the task definition, so it has this logic for "do I need to run setUp?" that checks the image tag. We don't need that for DO.